File: aoc2018/day1/day1.js
const fs = require('fs')

inputs = fs.readFileSync('inputs.txt')
    .toString()
    .trim()
    .split('\n')
    .map(i => +i);

// puzzle 1
console.log(inputs.reduce((a,b) => a+b, 0))

// puzzle 2
seen = new Set()
f = 0
for (;;) {
    inputs.forEach(input => {
        f += input
        if (seen.has(f)) {
            console.log(f)
            process.exit()
        }
        seen.add(f)
    });
}

File: aoc2018/day1/day1.py
inputs = open('inputs.txt').read().strip().split('\n')

# puzzle 1
print(sum([int(i) for i in inputs]))

# puzzle 2
seen = set()
f = 0
while True:
    for i in inputs:
        f += int(i)
        if f in seen:
            print(f)
            quit()
        seen.add(f)
